src/parser.ts

```diff
@@ -112,10 +125,12 @@ export function byNode<T extends ParserObject>(node: HTMLElement): T {
 /**
  * The public API for retrieving a parser object by ID
  * @param  {string}                 id The ID of the parser object to retrieve
+ * @param  {Document}               doc The optional Document. Defaults to `document`.
  * @return {T extends ParserObject}    The instance associated with the ID or `undefiend`
  */
-export function byId<T extends ParserObject>(id: string): T {
-	return <T> idMap[id];
+export function byId<T extends ParserObject>(id: string, doc: Document = document): T {
```

Comment:
I know we are doing the right thing on the ordering of the "optional" argument last, but I wonder from a developer ergonomics if an overloaded function might not be in order here... (this is more a of discussion/question on my part versus a strong opinion):

```typescript
function isDocument(value: any): value is document {
    return value instanceof Document;
}

export function byId<T extends ParserObject(id: string): T;
export function byId<T extends ParserObject(doc: Document, id: string): T;
export function byId<T extends ParserObject(doc: Document | string, id?: string): T {
    const idMap = isDocument(doc) ? idDocumentMap.get(doc) : idDocumentMap.get(document);
    return idMap && <T> idMap[id || doc];
}
```